src/components/progress/progress.a11y.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/// <reference types="../../../cypress/support"/>
+
+import React from 'react';
+import { EuiProgress } from './progress';
+import { EuiSpacer } from '../spacer';
+
+const ProgressCommonProps = {
+  color: 'success',
+  max: 100,
+};
+
+const ProgressBars = () => {
+  return (
+    <>
+      <div data-test-subj="cy-progress-1">
+        <EuiProgress valueText size="xs" value={0} {...ProgressCommonProps} />
+      </div>
+
+      <div data-test-subj="cy-progress-2">
+        <EuiProgress valueText size="s" value={33} {...ProgressCommonProps} />
+      </div>
+
+      <div data-test-subj="cy-progress-3">
+        <EuiProgress valueText size="m" value={66} {...ProgressCommonProps} />
+      </div>
+
+      <div data-test-subj="cy-progress-4">
+        <EuiProgress valueText size="l" value={100} {...ProgressCommonProps} />
+      </div>
+
+      <div data-test-subj="cy-progress-5">
+        <EuiProgress
+          valueText
+          label="Basic percentage"
+          size="l"
+          value={100}
+          {...ProgressCommonProps}
+        />
+      </div>
+
+      <EuiSpacer size="m" />
+
+      <div data-test-subj="cy-progress-6">
+        <EuiProgress valueText size="l" color="success" />
+      </div>
```

Comment:
super optional naming suggestion - it didn't jump out to me initially that this was the infinite loading example, it might be nice to use descriptive hooks
```suggestion
      <div data-test-subj="cy-progress-infinite">
        <EuiProgress valueText size="l" color="success" />
      </div>
```